File: tests/fixtures/sample-go-repo/internal/service/service.go
// Package service provides the sample domain service.
package service

import (
	"errors"
	"fmt"
)

// ValidationError is a domain-specific error for input validation failures.
type ValidationError struct {
	Field   string
	Message string
}

// Error implements the error interface.
func (e *ValidationError) Error() string {
	return fmt.Sprintf("validation failed on %s: %s", e.Field, e.Message)
}

// SampleService performs sample operations.
type SampleService struct {
	maxValue int
}

// NewSampleService creates a new SampleService with the given max.
func NewSampleService(max int) *SampleService {
	return &SampleService{maxValue: max}
}

// DoWork validates the input and returns double its value.
// Returns a ValidationError if the input is out of range.
func (s *SampleService) DoWork(input int) (int, error) {
	if input < 0 {
		return 0, &ValidationError{
			Field:   "input",
			Message: fmt.Sprintf("must be >= 0 but was %d. Suggested fix: pass a positive integer", input),
		}
	}
	if input > s.maxValue {
		return 0, &ValidationError{
			Field:   "input",
			Message: fmt.Sprintf("exceeds max %d. Try a value under %d", s.maxValue, s.maxValue),
		}
	}
	return input * 2, nil
}

// ParseConfig reads configuration from the given path.
func ParseConfig(path string) (map[string]string, error) {
	if path == "" {
		return nil, errors.New("path must not be empty. Check your config file location")
	}
	return nil, fmt.Errorf("config file %s not found. See docs/configuration.md for setup", path)
}

// internalHelper is an unexported helper function.
func internalHelper() int {
	return 42
}

type internalWorker struct {
	name string
}

File: tests/fixtures/sample-rust-repo/src/service.rs
use std::fmt;

/// A domain-specific validation error.
pub struct ValidationError {
    pub field: String,
    pub message: String,
}

impl fmt::Display for ValidationError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "validation failed on {}: {}", self.field, self.message)
    }
}

/// The sample service that performs operations.
pub struct SampleService {
    max_value: i32,
}

impl SampleService {
    /// Creates a new SampleService with the given maximum value.
    pub fn new(max: i32) -> Self {
        SampleService { max_value: max }
    }

    /// Validates the input and returns double its value.
    ///
    /// Returns a ValidationError if the input is out of range.
    pub fn do_work(&self, input: i32) -> Result<i32, ValidationError> {
        if input < 0 {
            return Err(ValidationError {
                field: "input".to_string(),
                message: format!(
                    "must be >= 0 but was {}. Suggested fix: pass a positive integer",
                    input
                ),
            });
        }
        if input > self.max_value {
            return Err(ValidationError {
                field: "input".to_string(),
                message: format!("exceeds max {}. Try a value under {}", self.max_value, self.max_value),
            });
        }
        Ok(input * 2)
    }
}

/// Parses configuration from the given path.
pub fn parse_config(path: &str) -> Result<(), String> {
    if path.is_empty() {
        panic!("path must not be empty. Check your config file location");
    }
    Ok(())
}

fn internal_helper() -> i32 {
    42
}

struct InternalWorker {
    name: String,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_do_work_positive_input_returns_double() {
        let svc = SampleService::new(1000);
        assert_eq!(svc.do_work(5).unwrap(), 10);
    }

    #[test]
    fn test_do_work_negative_input_returns_error() {
        let svc = SampleService::new(1000);
        assert!(svc.do_work(-1).is_err());
    }

    #[test]
    fn test_parse_config_empty_panics() {
        let result = std::panic::catch_unwind(|| parse_config(""));
        assert!(result.is_err());
    }
}
